language-design.md

```diff
@@ -106,7 +209,25 @@ you do so? If not, why not?
 
 **Response**
 
+The quote that we already cited above is very relevant:
+
+> If your standard is natural language, division/fraction notation isn't particularly intuitive.
+> But to someone who knows the notation, ab / b is very intuitive, and it's immediately obvious that
+> you can cancel the bs. So the goal of an intuitive programming syntax shouldn't be to emulate
+> natural language, necessarily, but to provide intuitive hooks and syntax for the types of things
+> that programmers do a lot. 
+> [Comment by "Peter" ([Source](https://www.fastcodesign.com/1665735/why-arent-computer-programming-languages-designed-better#comment-bc08f700-3c06-11e1-921b-9794fc737ae8))] [Pavlus, 2012]
+
+There are (at least) two main considerations for syntax. The first is how well it fits the task at hand, *in a vacuum*. That is, how good it is considering only the language and the task, and not what users are already used to or know. The other is how easy it will be for real users to adopt. Considering this, the analogy with mathematics is apt. After all, mathematicians do *not* use natural language. They use mathematical notation, because
+it is simply better for the task. However, when new ideas come along, to begin with, the ideas are explained using natural language and older notation. The optimal descriptions for learning concepts are not the same as the optimal descriptions for using them.
 
+This mirrors the tradeoffs in language design pretty well. Ultimately, notation specially crafted for the task really is better, and not just for efficiency's sake. The price is the learning curve. There is also the risk of a syntax not becoming widely adopted and therefore not being suitable for discussing ideas with other people. The idea of extensible languages is somewhat of a solution to this. New concepts are first described in terms of primitives people already understand, but the universal language is flexible enough to allow new and more specialized notation to be added in beyond that point. This is how mathematics itself handles the problem.
+
+The priority should probably be on how well the language design fits the actual task, on how good it is for people to use rather than to learn. Within that realm, you should optimize for familiarity. This means that, where suitable domain-specific notation and concepts exist, they should be used. This includes natural language, so if there is a natural language way to write something that doesn't really have any drawbacks, it should be preferred. Thus, for example, you should prefer Haskell's "if .. then .. else .." to Java's ".. ? .. : ..".
```

Comment:
I agree with the caveat that there are some concepts and notations that are ambiguous, and those should be avoided or strictly defined so that users are not confused.